controller/app.controller.js

```diff
@@ -20,13 +20,50 @@ const wrongPath = (req, res, next) => {
 };
 
 const getArticleById = (req, res, next) => {
-    selectArticle(req.params.id)
+    selectArticleById(req.params.id)
         .then((data) => {
-            res.status(200).send({ rows: data });
+            res.status(200).send({ article: data[0] });
+            })
+        .catch((err) => {
+            next(err);
+        });
+}
+
+const getAllArticles = (req, res, next) => {
+    selectArticles()
+        .then((data) => {
+            res.status(200).send({ articles: data });
         })
         .catch((err) => {
             next(err);
         });
 };
 
-module.exports = { getAllTopics, getApis, wrongPath, getArticleById };
+const getComments = (req, res, next) => {
+    selectAllCommentsForArticle(req.params.article_id)
+    .then((data) => {
+        if (data.length === 0){
```

Comment:
As per my comments on your previous pr - we need to check why this is empty. If it is down to invalid ID, we need to throw and error, otherwise just return an empty array as it could be down to just no comments on valid article. 🙂